sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -633,6 +628,12 @@ export class ApplicationService extends RecordDataModelService<Application> {
         applicationNumber: true,
         pirDeniedOtherDesc: true,
         submittedDate: true,
+        precedingApplication: options?.loadPrecedingApplication
+          ? {
+              id: true,
+              data: true as unknown,
+            }
+          : null,
```

Comment:
I would recommend to fetch the id of preceding application always and get the dynamic data based on the flag loadDynamicData which is available already in the method. 

![image](https://github.com/user-attachments/assets/4a3c70d2-be39-4ab2-843b-9f4c881c9699)

Reply:
By doing so as per the comment https://github.com/bcgov/SIMS/pull/4340/files#r1953508167

We can remove the method `getLastApplicationVersion()` completely